src/differentials/notimplemented.jl

```diff
@@ -0,0 +1,80 @@
+"""
+    NotImplemented
+
+This differential indicates that the derivative is not implemented.
+"""
+struct NotImplemented{M,S,I} <: AbstractDifferential
```

Comment:
Now that we don't replace things with `nothing` if not in debug mode this probably doesn't need to be type parametric any more.
Even the info can probably be defaulted to the empty string.
(or maybe better; made required)

(This applies also to the `NotImplementedException`)

Reply:
Good point, I removed the type parameter and made the debugging information mandatory.